gordo_components/builder/build_model.py

```diff
@@ -326,13 +329,39 @@ def set_seed(self, seed: int):
         np.random.seed(seed)
         random.seed(seed)
 
+    @staticmethod
+    def build_split_dict(X: pd.DataFrame, split_obj: Type[BaseCrossValidator]) -> dict:
+        """
+        Get dictionary of cross-validation training dataset split metadata
+
+        Parameters
+        ----------
+        X: pd.DataFrame
+            The training dataset that will be split during cross-validation.
+        split_obj: Type[sklearn.model_selection.BaseCrossValidator]
+            The cross-validation object that returns train, test indices for splitting.
+
+        Returns
+        -------
+        split_metadata: Dict[str,Any]
+            Dictionary of cross-validation train/test split metadata
+        """
+        split_metadata: Dict[str, Any] = dict()
+        for i, (train_ind, test_ind) in enumerate(split_obj.split(X)):
+            for name, ind in zip(["train", "test"], [train_ind, test_ind]):
+                for pos, idx in zip(["start", "end"], [0, -1]):
+                    split_metadata.update(
+                        {f"fold-{i+1}-{name}-{pos}": X.index[ind[idx]]}
+                    )
```

Comment:
I like the params, but by the third for loop, my brain gets slightly tired.
Is something like this easier on the eyes? 
```python
split_metadata.update(
    {
        f"fold-{i+1}-train-start": X.index[train_ind[0]],
        f"fold-{i+1}-train-end": X.index[train_ind[-1]],
        f"fold-{i+1}-test-start": X.index[test_ind[0]],
        f"fold-{i+1}-test-end": X.index[test_ind[-1]]
    }
)
```

Reply:
That's what I done did this first time, so I dooo agree. Should I change it back? I think both are reasonably readable on that note, though. 

I just generalized things while I abstracted it out to a function.

Reply:
I find the nested loops has my brain doing a lot of extra work remapping names to variables. Primarily because what is being iterated over isn't a 'big' iterator, just two values. :man_shrugging:

Reply:
Ok, changed it back now :+1: